src/utils/relayChainApi/transfer.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  CollateralUnit,
+  DefaultTransactionAPI
+} from '@interlay/interbtc-api';
+import {
+  Currency,
+  MonetaryAmount
+} from '@interlay/monetary-js';
+import { web3FromAddress } from '@polkadot/extension-dapp';
+import { AddressOrPair } from '@polkadot/api/types';
+import { ApiPromise } from '@polkadot/api';
+
+import { RELAYCHAIN_ID } from '../../constants';
+import { createRelayChainApi } from './createRelayChainApi';
+
+type XCMTransferAmount = MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>;
+
+const createDest = (api: ApiPromise) => {
+  const x1 = api.createType('XcmV1Junction', { parachain: RELAYCHAIN_ID });
+  const interior = api.createType('XcmV1MultilocationJunctions', { x1 });
+  const v1 = api.createType('XcmV1MultiLocation', { parents: 0, interior });
+
+  return api.createType('XcmVersionedMultiLocation', {
+    v1
+  });
+};
+
+const createBeneficiary = (api: ApiPromise, id: string) => {
+  const network = api.createType('XcmV0JunctionNetworkId', { any: true });
+  const x1 = api.createType('XcmV1Junction', {
+    accountId32: {
+      network,
+      id
+    }
+  });
+  const interior = api.createType('XcmV1MultilocationJunctions', { x1 });
+  const v1 = api.createType('XcmV1MultiLocation', { parents: 0, interior });
+
+  return api.createType('XcmVersionedMultiLocation', {
+    v1
+  });
+};
+
+const createAssets = (api: ApiPromise, transferAmount: XCMTransferAmount) => {
+  const fungible = transferAmount.toString(transferAmount.currency.rawBase);
+  const fun = api.createType('XcmV1MultiassetFungibility', { fungible });
+  const interior = api.createType('XcmV1MultilocationJunctions', { here: true });
+  const concrete = api.createType('XcmV1MultiLocation', { parents: 0, interior });
+  const id = api.createType('XcmV1MultiassetAssetId', { concrete });
+  const v1 = api.createType('XcmV1MultiassetMultiAssets', [{ id, fun }]);
+
+  return api.createType('XcmVersionedMultiAssets', {
+    v1
+  });
+};
+
+// Originating account is passed into avoid creating a dependency on the interBTC api instance
+const xcmTransfer = async (
+  originatingAccount: AddressOrPair,
+  destinationAddress: string,
+  transferAmount: XCMTransferAmount
+): Promise<void> => {
+  const api = await createRelayChainApi();
+
+  if (!api) return;
+
+  // Create transaction api instance on the relaychain
+  const transactionApi = new DefaultTransactionAPI(api, originatingAccount);
+
+  // Signer needs to be set explicitly for the new api
+  const { signer } = await web3FromAddress(originatingAccount.toString());
+  api.setSigner(signer);
```

Comment:
I'm surprised that you need this after constructing the `transactionApi` with `originatingAccount`. This part should be handled automatically by `signAndSend`
https://github.com/interlay/interbtc-api/blob/master/src/parachain/transaction.ts#L59

Does the tx fail if this is removed?

Reply:
I've added a comment so we can look into it properly when we have a little more time

Reply:
Ok I figured it out - you either pass a KeyringPair object (derived from a mnemonic), or you pass an AccountId along with a signer. So this code is correct, since it uses the extension to sign. This is the lib code: https://github.com/interlay/interbtc-api/blob/master/src/interbtc-api.ts#L149

And this is the polkadot-js docs for:
- signing with the extension: https://polkadot.js.org/docs/extension/usage/
- signing with the seed: https://polkadot.js.org/docs/extension/usage/

Reply:
Ah nice :) do we want to change this down the line to match the lib implementation, or is this ok as it is?

Reply:
It's ok as it is, this is very much in flux still anyway